web/02a-createsite.php

```diff
@@ -0,0 +1,247 @@
+<?php
+/**
+ * Copyright (c) 2012 University of Illinois, NCSA.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the 
+ * University of Illinois/NCSA Open Source License
+ * which accompanies this distribution, and is available at
+ * http://opensource.ncsa.illinois.edu/license.html
+ */
+
+// Check login
+require("common.php");
+open_database();
+if ($authentication) {
+  if (!check_login()) {
+    header( "Location: index.php");
+    close_database();
+    exit;
+  }
+  if (get_page_acccess_level() > $min_run_level) {
+    header( "Location: history.php");
+    close_database();
+    exit;
+  }
+}
+
+# parameters
+if (!isset($_REQUEST['lat'])) {
+  die("Need latitude.");
+}
+$lat=$_REQUEST['lat'];
+if (!isset($_REQUEST['lon'])) {
+  die("Need longitude.");
+}
+$lon=$_REQUEST['lon'];
+
+?>
+<!DOCTYPE html>
+<html>
+<head>
+<title>PEcAn Create Site</title>
+<link rel="shortcut icon" type="image/x-icon" href="favicon.ico" />
+<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no" />
+<meta http-equiv="content-type" content="text/html; charset=UTF-8" />
+<link rel="stylesheet" type="text/css" href="sites.css?id=<?php echo rand(); ?>" />
+<script type="text/javascript" src="jquery-1.10.2.min.js"></script>
+<script type="text/javascript">
+  function validate() {
+    $("#error").html("&nbsp;");
+    if ($("#newsite").val() == "") {
+      $("#error").html("Enter sitename to continue");
+      return false;
+    }
+    // if ($("#city").val() == "") {
+    //   $("#error").html("Enter city to continue");
+    //   return false;
+    // }
+    // if ($("#state").val() == "") {
+    //   $("#error").html("Enter state to continue");
+    //   return false;
+    // }
+    if ($("#county").val() == "") {
+      $("#error").html("Enter county to continue");
+      return false;
+    }
+    // if ($("#timezone").val() == "") {
+    //   $("#error").html("Enter timezone to continue");
+    //   return false;
+    // }
+    if ($("#lat").val() == "") {
+      $("#error").html("Enter lat to continue");
+      return false;
+    }
+    if ($("#lon").val() == "") {
+      $("#error").html("Enter lon to continue");
+      return false;
+    }
+    // if ($("#elevation").val() == "") {
+    //   $("#error").html("Enter elevation to continue");
+    //   return false;
+    // }
+    if ($("#temperature").val() == "") {
```

Comment:
The current Add Site interface requires a whole lot of information that's optional (temp, precip, clay, sand, county). Also, the interface doesn't give any hints about which fields are required and which are optional. I'd recommend adding a * or BOLD to the required variables.

Reply:
only sitename, lat, lon is now required.